pkg/querier/vcs/client/encryption.go

```diff
@@ -24,11 +25,19 @@ func encryptToken(token *oauth2.Token, key []byte) (string, error) {
 	return base64.StdEncoding.EncodeToString(enc), nil
 }
 
+// TODO(bryan) this is kind of a hack and I found it through debugging.
+const gcmNonceSize = 12
+
 func decryptToken(encodedText string, key []byte) (*oauth2.Token, error) {
 	encryptedData, err := base64.StdEncoding.DecodeString(encodedText)
 	if err != nil {
 		return nil, err
 	}
+
+	if len(encryptedData) < gcmNonceSize {
```

Comment:
I suspect we need to recover panic in this function or above too just in case ?

Reply:
Yeah, we should. Shouldn't all recoverable panics get caught by the grpc handler? Why doesn't this happen today?